ramp4-pcar4/ramp4-pcar4
# details panel re-work

### Related Item(s)
#2091 

### Changes
- Completely re-wrote the details fixture to use only a single panel.
- Added a new sample page to test a teleported details fixture

### Notes
Please test everything against the existing RAMP4 details fixture so we can ensure I haven't missed or broken anything!

### Testing
I've listed a few specific sample pages below with different scenarios (steps should apply to all of them).

Steps:
1. Click on a point on the map and ensure that the panel opens correctly and the information appears as expected.
2. Clicking on a spot with multiple points from the same layer should display the paginator and allow you to switch between points.
3. If clicking on a spot with multiple points from different layers, the count should be reflected in the sidebar on the left.
4. Ensure the highlight toggle is working correctly
5. Ensure that you can open data points from the grid.
6. Test that the fixture is keyboard accessible: <br /> - If you tab into the sidebar it should expand automatically, and you should be able to switch layers using the arrow keys.<br /> - You should also be able to use the arrow keys to navigate between individual points in the list view.

**Sample Pages**
Map with multiple layers (sidebar should appear): https://ramp4-pcar4.github.io/ramp4-pcar4/details-rework/demos/index-samples.html?sample=1
Map with a single layer (sidebar shouldn't appear): https://ramp4-pcar4.github.io/ramp4-pcar4/details-rework/demos/index-samples.html?sample=20
Layer with custom details template: https://ramp4-pcar4.github.io/ramp4-pcar4/details-rework/demos/index-samples.html?sample=25
Teleported details fixture: https://ramp4-pcar4.github.io/ramp4-pcar4/details-rework/demos/index-teleport-details.html


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ramp4-pcar4/ramp4-pcar4/2119)
